sdk/python/kfp/dsl/types.py

```diff
@@ -94,6 +98,35 @@ class InconsistentTypeException(Exception):
 	'''InconsistencyTypeException is raised when two types are not consistent'''
 	pass
 
+
+class InconsistentTypeWarning(Warning):
+	'''InconsistentTypeWarning is issued when two types are not consistent'''
+	pass
+
+
+def verify_type_compatibility(given_type, expected_type, error_message_prefix : str = ''):
```

Comment:
Should all parameters have a defined type?

Reply:
Fixed in https://github.com/kubeflow/pipelines/pull/1934